src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
And, yes, that means we're supporting something that doesn't directly inherit from an interface we're requesting, and instead are duck typing, and haven't taken the time to do some ABC tricks to make it airtight, but it's straightforward enough that in offline discussion we determined it was desirable to implicitly support it.